app/Testing/GuzzleTapper.php

```diff
@@ -105,24 +105,22 @@ public function replaceMatchFile(string $method, string $urlPattern, string $fil
 
     /**
      * Given a RequestInterface (from a unit test trying to make a Guzzle request)
-     * Find the first appropriate match (NOT best-match!) in $this->matches
      *  - Log that the request happened (so we can assert later what requests happened in what order)
+     *    - Even if the request is not matched, it will be logged so you can have confidence in your tests
+     *  - Find the first appropriate match (NOT best-match!) in $this->matches
      *  - Run and return the behavior described in the match
      * @returns Response a Guzzle response object, including status, headers, body
      * @returns Exception if the behavior *returns* an exception, Guzzle will throw it, e.g. a BadResponseException
      * @throws OutOfBoundsException if the RequestInterface does not match any $this->matches
      */
     public function response(RequestInterface $request, array $options = [])
     {
-        $requestMethod = $request->getMethod();
-        if (!array_key_exists($requestMethod, $this->matches)) {
-            return new \OutOfBoundsException("No responses match method {$requestMethod}");
```

Comment:
The new null-coalesce on new line 122 means we don't have to maintain separate exceptions with slightly different messages. The message on new line 131 includes the full URL, so you're not forced to figure out "what uses a GET that I forgot to mock?"